disaster_recovery/restore/disaster_recovery.py

```diff
@@ -0,0 +1,621 @@
+            DeletionProtection=False,
+            DBClusterIdentifier=self.db_cluster_identifier_source,
+            ApplyImmediately = True
+        )
+
+        self.command_response('Drop protection', response)
+
+        self.wait_on_cluster_available(self.db_cluster_identifier_source)
+
+    def get_instance_info(self):
+        response = self.client.describe_db_instances(
+            Filters=[{'Name': 'db-cluster-id', 'Values': [self.db_cluster_identifier_source]}],
+        )
+
+        for instance in response["DBInstances"]:
+            instance_obj = {
+                "DBInstanceIdentifier": instance["DBInstanceIdentifier"],
+                "DBInstanceClass": instance["DBInstanceClass"],
+                "AvailabilityZone": instance["AvailabilityZone"]
+            }
+            self.instances.append(instance_obj)
+
+        print(self.instances)
+
+    def get_cluster_info(self):
+        response = self.client.describe_db_clusters(
+            DBClusterIdentifier=self.db_cluster_identifier_source
+        )
+        print(response)
+        self.AllocatedStorage = response['DBClusters'][0]['AllocatedStorage']
+        self.AvailabilityZones = response['DBClusters'][0]['AvailabilityZones']
+        self.BackupRetentionPeriod = response['DBClusters'][0]['BackupRetentionPeriod']
+        self.db_cluster_identifier_source = response['DBClusters'][0]['DBClusterIdentifier']
+        self.DBClusterParameterGroup = response['DBClusters'][0]['DBClusterParameterGroup']
+        self.DBSubnetGroup = response['DBClusters'][0]['DBSubnetGroup']
+        self.MultiAZ = response['DBClusters'][0]['MultiAZ']
+        self.Engine = response['DBClusters'][0]['Engine']
+        self.EngineVersion = response['DBClusters'][0]['EngineVersion']
+        self.VpcSecurityGroups = self.format_list(
+            response['DBClusters'][0]['VpcSecurityGroups'], 'VpcSecurityGroupId'
+        )
+        self.StorageEncrypted = response['DBClusters'][0]['StorageEncrypted']
+        self.KmsKeyId = response['DBClusters'][0]['KmsKeyId']
+        self.Capacity = None if 'Capacity' not in response['DBClusters'][0] else response['DBClusters'][0]['Capacity']
+        self.EngineMode = response['DBClusters'][0]['EngineMode']
+        self.DeletionProtection = response['DBClusters'][0]['DeletionProtection']
+        self.TagList = response['DBClusters'][0]['TagList']
+        self.AutoMinorVersionUpgrade = response['DBClusters'][0]['AutoMinorVersionUpgrade']
+        self.HttpEndpointEnabled = response['DBClusters'][0]["HttpEndpointEnabled"]
+
+    def restore_from_snapshot(self):
+        if self.same_target:
+            self.db_cluster_identifier_target = f"{self.db_cluster_identifier_source}-temp"
+
+        self.restore_cluster_ss()
+        if self.EngineMode != 'serverless':
+            self.create_db_instances()
+
+        if self.same_target:
+            self.overwrite_existing_cluster()
+
+    def restore_cluster_ss(self):
```

Comment:
What does `ss` stand for? Could we get it expanded to the full words?

Reply:
Yeah not sure why i did that. Very naughty.